components/Dashboard/Post/PostAuthorCard.tsx

```diff
@@ -26,6 +32,37 @@ const PostAuthorCard: React.FC<PostAuthorCardProps> = ({ author }) => {
     languagesNative.push(language)
   }
 
+  const hasFollowedAuthor =
+    currentUser.following.find((user: UserType) => user.id === author.id) !== undefined
+
+  const [followUserMutation, { loading: followLoading }] = useFollowUserMutation({
+    onCompleted: () => {
+      refetch()
+    },
+  })
+
+  const handleFollowUser = () => {
+    followUserMutation({
+      variables: {
+        followedUserId: author.id,
+      },
+    })
+  }
+
+  const [unfollowUserMutation, { loading: unfollowLoading }] = useUnfollowUserMutation({
+    onCompleted: () => {
+      refetch()
```

Comment:
This is currently not working as expected, and I'm noticing some flickering upon clicking this but also just page refresh